### PR TITLE
Restructuring the folders

### PR DESCRIPTION
**Pull request by JamesABaker**
# Improving the default file structure

Defaults matter. By defining a structure in the template we can nudge people to working in a certain way.

This approach nudges people to work in a package focussed way, but provides flexibility for projects that are script or notebook intensive.

Whilst a data folder exists, data itself is gitignored, and generally we would want people to be interacting with data on s3.

## Description

Added [cookiecutter-data-science](https://github.com/drivendataorg/cookiecutter-data-science/tree/master/%7B%7B%20cookiecutter.repo_name%20%7D%7D) style folder structure with data/, models/, notebooks/, reports/, src/, and scripts/ directories, comprehensive .gitignore for data files, and READMEs explaining where code belongs.

## Checklist for submission

- [x] GitHub Actions are passing
- [x] I have added relevant tests (or explain why tests aren't needed)
- [x] I have considered if an ADR is required for any architectural decisions
- [x] Documentation has been updated if needed (README, comments, etc.)
- [x] The PR does not exceed 500 lines of code (excluding config files or data changes)

## To the reviewer

When doing a review, please follow [the GDS approach](https://gds-way.digital.cabinet-office.gov.uk/manuals/code-review-guidelines.html#how-to-review-code).

For this PR, please focus on:

- [ ] **Code Audit**. A comprehensive review according to the [ONS DuckBook standards](https://best-practice-and-impact.github.io/qa-of-code-guidance/checklist_lower.html#template-checklist)
- [ ] **Peer Review**. A review of the process, method, or maths being sensible
- [x] **Code Review**. A code review for:
    - code implementation (does the code do what it says it does)
    - **documentation (is it easy to understand what the code is trying to do)**
    - tests (is the code being tested for edge cases, unexpected inputs, and normal operation)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -51,8 +51,40 @@ wheels/
...
 
-# Data files (add specific patterns as needed)
-# *.csv
-# *.xlsx
-# *.parquet
-# data/
+# Data files - common formats
+*.csv
+*.xlsx
+*.xls
```
Comment (zslade) [inline .gitignore]: Perhaps could include `.xlsm`, `.docx`, `.doc` and `.pdf` as well, to be extra "safe"

File: .gitignore [hunk 2]
```diff
@@ -51,8 +51,40 @@ wheels/
...
+*.rda
+*.RData
+
+# Data directories (except .gitkeep files and READMEs)
+data/*
+!data/.gitkeep
+!data/README.md
+!data/*/
+data/*/*
+!data/*/.gitkeep
```
Comment (zslade) [inline .gitignore]: I think this could be shortened to:
```suggestion
data/**/* # Ignore all files in data directory recursively
!data/**/*.gitkeep # But keep all .gitkeep files
!data/**/README.md # And keep all README.md files
```
which also safeguards against data being put into directories at any depth

File: README.md [hunk 3]
```diff
@@ -190,12 +209,16 @@ If you're working with R:
 
 Edit this README.md file to document your project accurately. Take the time to create a clear, engaging, and informative README.md file. Include information like what your project does, how to install and run it, how to contribute, and any other pertinent details.
 
+Also make sure the badge urls are correct for your repository:
```
Comment (zslade) [inline README.md]: Minor suggestion
```suggestion
Also make sure the badge urls are correct for your repository, e.g.:
```

File: README.md [hunk 4]
```diff
@@ -43,9 +39,46 @@ This template provides a robust foundation for data science projects:
...
+    ├── integration/
+    └── e2e/          # End-to-end tests
+```
+
+See individual directories for detailed READMEs. Key points:
+
+- **data/**, **models/**, **reports/**: Gitignored to prevent committing large files
+- **src/**: Installable as package with `pip install -e .` (see [src/README.md](src/README.md))
+- **scripts/**: One-off executables that use `src/` code (see [scripts/README.md](scripts/README.md))
+- **notebooks/**: Use numbered prefixes (e.g., `01-exploration.ipynb`)
```
[the remark this answers is not in the document]
Comment (JamesABaker) [reply README.md]: I think this is the only case when it comes to python files, but do you mean extending this to things like requirements-dev.txt?
